.github/workflows/ci.yml

```diff
@@ -0,0 +1,42 @@
+# This workflow will build the project
+
+name: CI
+
+on:
+  push:
+    branches:
+      - '3.0'
+    tags:
+      - v3.*
+  pull_request:
+    branches:
+      - '3.0'
+
+  # Allow running this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Gradle wrapper validation
+        uses: gradle/wrapper-validation-action@v1
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Build with Gradle
+        run: |
+          echo "Running build for commit ${{ github.sha }} in ${{ github.head_ref }}"
+          ./gradlew build -x javadoc
```

Comment:
Should this line be calling a modified version of `buildViaTravis.sh`?

Reply:
This PR is only to get the CI action working in GitHub. I will create another one for the publishing action.